service/src/main/java/com/derpgroup/derpwizard/dao/impl/H2EmbeddedAccountLinkingDAO.java

```diff
@@ -0,0 +1,407 @@
+    }
+    return null;
+  }
+  
+  protected ResultSet executeQuery(String sql){
+    return executeQuery(sql, null);
+  }
+  
+  protected ResultSet executeQuery(String sql, ArrayList<? extends Object> parameters){
+
+    CachedRowSet crs = null;
+    
+    try(
+        PreparedStatement unpreparedStatement = conn.prepareStatement(sql);
+        PreparedStatement statement = prepStatement(unpreparedStatement, parameters);
+        ResultSet rs = statement.executeQuery();
+        ){
+      crs = RowSetProvider.newFactory().createCachedRowSet();
+      crs.populate(rs);
+      return crs;
+    }catch(SQLException e){
+      e.printStackTrace();
+    }
+    return null;
+  }
+  
+  protected void setupFixtureData() throws SQLException{
+    conn.setAutoCommit(false);
+    String userTableCreation = "CREATE TABLE User(id varchar(255) PRIMARY KEY NOT NULL,"
+        + "firstName varchar(255) NULL,"
+        + "dateCreated TIMESTAMP NOT NULL DEFAULT(NOW()));";
+    executeStatement(userTableCreation);
+    
+    String accountLinkTableCreation = "CREATE TABLE AccountLink("
+        + "userId varchar(255) NOT NULL,"
+        + "externalUserId varchar(255) NULL,"
+        + "externalSystemName varchar(64) NOT NULL,"
+        + "externalSystemToken varchar(255) NULL,"
+        + "externalSystemRefreshToken varchar(255) NULL"
+        + ");";
+    executeStatement(accountLinkTableCreation);
+    
+    String linkingTokenTableCreation = "CREATE TABLE LinkingToken("
+        + "token UUID NOT NULL DEFAULT(RANDOM_UUID()),"
+        + "userId varchar(255) NOT NULL,"
+        + "dateCreated TIMESTAMP NOT NULL DEFAULT(NOW())"
+        + ");";
+    executeStatement(linkingTokenTableCreation);
+    
+    String authorizationTableCreation = "CREATE TABLE Authorization("
+        + "token UUID NOT NULL DEFAULT(RANDOM_UUID()),"
+        + "userId varchar(255) NOT NULL,"
+        + "dateCreated TIMESTAMP NOT NULL DEFAULT(NOW())"
+        + ");";
+    executeStatement(authorizationTableCreation);
+    conn.commit();
+    conn.setAutoCommit(true);
+  }
+  
+  protected Connection getConn(){
+    return conn;
+  }
+
+  @Override
+  public UserAccount getUserByUserId(String alexaUserId) {
+    String userSelect = "SELECT id, firstName FROM User WHERE id = ?;";
+    ArrayList<String> parameters = new ArrayList<String>();
+    parameters.add(alexaUserId);
+    ResultSet response = executeQuery(userSelect, parameters);
+    
+    UserAccount user = new UserAccount();
+    try {
+      if(!response.first()){
+        return null;
+      }
+      user.setUserId(response.getString("id"));
+      user.setFirstName(response.getString("firstName"));
+    } catch (SQLException e) {
+      e.printStackTrace();
+    }
+    return user;
+  }
+
+  @Override
+  public UserAccount updateUser(UserAccount user) {
+
+    String userCreate = "MERGE INTO User(id, firstName) KEY(id) VALUES(?,?);";
+    
+    ArrayList<String> parameters = new ArrayList<String>();
```

Comment:
List<String> parameters = new ArrayList<>();

People at work always jump on me for not doing this. So now I'm punishing you with it.